### ISSUE TITLE
close button not working

### ISSUE THREAD

**Issue by dookki** [lencx/ChatGPT#302]
### Error message

i can't close the app after opening it, the close button works like Minimize rather than close, I have to use taskmanager on the window to close the app.

### OS

windows 11

**Comment by lencx** [comment]
It's a feature, not a bug, because it's targeted for high frequency use.

**Comment by dookki** [comment]
> Đó là một tính năng, không phải lỗi, vì nó được nhắm mục tiêu để sử dụng với tần suất cao.
One thing I wonder is that there is already a Minimize feature, why close is minimize again
![image](https://user-images.githubusercontent.com/88790001/216538223-a1a211e6-9e27-476d-9096-2337ba5fa0a6.png)

**Comment by lencx** [comment]
Because I can't remove any of the three buttons, but I can rewrite its original event.

**Comment by heldheld** [comment]
In the 0.8 update we did have the option to close to tray, so why is this gone?

**Comment by lencx** [comment]
The tray needs to be closed by configuration, not by the close button (`Preferences -> System Tray`).

**Comment by heldheld** [comment]
I don't want to close the tray, I want to close the application TO the tray. That means closing the application window (so not minimizing it) but keeping the tray icon. It's basically the same behaviour as using the Global Shortcut to 'hide' the window. This issue was fixed in 0.8 but somewhere in a later update got removed again.

**Comment by lencx** [comment]
> I don't want to close the tray, I want to close the application TO the tray. That means closing the application window (so not minimizing it) but keeping the tray icon. It's basically the same behaviour as using the Global Shortcut to 'hide' the window. This issue was fixed in 0.8 but somewhere in a later update got removed again.

Yes, that feature has been removed, so if you think it doesn't meet your requirements, we recommend building it locally yourself.

**Comment by heldheld** [comment]
> > I don't want to close the tray, I want to close the application TO the tray. That means closing the application window (so not minimizing it) but keeping the tray icon. It's basically the same behaviour as using the Global Shortcut to 'hide' the window. This issue was fixed in 0.8 but somewhere in a later update got removed again.
> 
> Yes, that feature has been removed, so if you think it doesn't meet your requirements, we recommend building it locally yourself.

Ah, okay, thanks for the clarification. May I ask why it has been removed?

I'd love to build it myself, I've no idea how to do that though unfortunately...

**Comment by Manamama** [comment]
Re:

> May I ask why it has been removed?

1. So that it runs at the background: it minimizes and not closes. 
2. You can always use pkill or its Windows equivalent. 
(Checks it in ChatGPT 🔢 : 

> In Windows, the equivalent command to pkill is the taskkill command. It can be used to terminate a process using its process ID (PID) or image name. 

)

> I'd love to build it myself, I've no idea how to do that though unfortunately...

1. git clone https://github.com/lencx/ChatGPT
2. Learn lots of code, e.g. from ChatGPT (it took me two or three days ;)
3. Tinker, (re)build and recompile ;)

**Comment by heldheld** [comment]
> 1. So that it runs at the background: it minimizes and not closes.

It does still run in the background when you close it though. Only when you completely quit the application it doesn't. A lot of applications work like this in Windows. And you don't need 2 buttons (minimize and close) that do the same thing, it's counter-intuitive. You could make a checkbox in settings that makes the close button act like the global shortcut and all would be perfect (assuming that this is possible), people could set it the way they prefer. Giving the user options to change the behaviour of buttons/features of an application to their liking always sounds like a good idea to me. Removing user choice is weird to me.

Anyway, I don't feel like this is an argument I'm going to win if I look at the update-cycle and comments by the developer, so I'll leave it now, I've made my point. Love the application and will definitely keep using it, just missing some basic quality of life features.

**Comment by lencx** [comment]
Yes, it doesn't meet your requirements and leaving is the best option.
